internal/build/tar_test.go

```diff
@@ -77,6 +77,35 @@ func TestLen(t *testing.T) {
 	}
 }
 
+func TestDontArchiveTiltfile(t *testing.T) {
+	f := newFixture(t)
+	ab := NewArchiveBuilder()
+	defer ab.close()
+	defer f.tearDown()
+
+	f.WriteFile("a", "a")
+	f.WriteFile("Tiltfile", "Tiltfile")
+
+	paths := []pathMapping{
+		pathMapping{
+			LocalPath:     f.JoinPath("a"),
+			ContainerPath: "/a",
+		},
+		pathMapping{
+			LocalPath:     f.JoinPath("Tiltfile"),
+			ContainerPath: "/Tiltfile",
+		},
+	}
+
+	err := ab.ArchivePathsIfExist(f.ctx, paths)
+	if err != nil {
+		f.t.Fatal(err)
+	}
+	actual := tar.NewReader(ab.buf)
+	f.assertFileInTar(actual, expectedFile{Path: "a", Contents: "a"})
+	f.assertFileInTar(actual, expectedFile{Path: "Tiltfile", Missing: true})
```

Comment:
this won't do what you expect, because assertFileInTar has already consumed the tarball, you need to use the multi-file version